### DISCUSSION TITLE
Redundant Outer Loop and Potential Bug in get_images Function

### DISCUSSION THREAD

**Discussion by 2ndpic** [comfyanonymous/ComfyUI#4650] [General]
[Link to the code](https://github.com/comfyanonymous/ComfyUI/blob/38c22e631ad090a4841e4a0f015a30c565a9f7fc/script_examples/websockets_api_example.py#L44C5-L52C51)

```python
for o in history['outputs']:
    for node_id in history['outputs']:
        node_output = history['outputs'][node_id]
        if 'images' in node_output:
            images_output = []
            for image in node_output['images']:
                image_data = get_image(image['filename'], image['subfolder'], image['type'])
                images_output.append(image_data)
        output_images[node_id] = images_output
```

First, I believe the outer loop `for o in history['outputs']:` is redundant and can be removed. This loop seems to serve no practical purpose as it merely iterates over the length of `history['outputs']` without performing any operations on each element.

Secondly, I think `images_output = []` should be placed above the condition `if 'images' in node_output:`. The current code structure will result in an error `local variable 'images_output' referenced before assignment` when the `if` condition is false.

Can the author or community members explain if there are any additional considerations for this code, or if it was simply written in error? Thank you!

**Suggested modification:**

```python
for node_id in history['outputs']:
    node_output = history['outputs'][node_id]
    images_output = []  # Initialize images_output here
    if 'images' in node_output:
        for image in node_output['images']:
            image_data = get_image(image['filename'], image['subfolder'], image['type'])
            images_output.append(image_data)
    output_images[node_id] = images_output
```

**Comment by ltdrdata** [comment]
I made a PR for this.
https://github.com/comfyanonymous/ComfyUI/pull/4656
